File: GraphicsEngine/Material.cpp
#include "stdafx.h"

#include "AllHeaders.h"

namespace GraphicsEngineSpace {

	Material::Material(String^name)
	{
		Name=name;

		Layers=gcnew List<MaterialLayer^>();
	}

	Material::~Material()
	{
		for each(MaterialLayer^ layer in Layers)
		{
			delete layer;
		}

		Layers->Clear();
	}


}

File: MaterialEngine/Material/Material.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AIOEngine.Graphics
{
    public class Material
    {
        public string Name { get; private set; }

        public List<MaterialLayer> Layers { get; private set; }

        public Material(string name)
        {
            Name = name;

            Layers = new List<MaterialLayer>();
        }

        public MaterialLayer CreateLayer()
        {
            MaterialLayer layer = new MaterialLayer();

            Layers.Add(layer);

            return layer;
        }
    }
}
